desk/app/channels.hoon

```diff
@@ -157,20 +176,141 @@
         =^subs:s
         =pimp:imp
     ==
+  ++  state-7-to-8
+    |=  s=state-7
+    ^-  state-8
+    s(- %8, v-channels (v-channels-7-to-8 v-channels.s))
+  ++  v-channels-7-to-8
+    |=  vc=v-channels:v7:old:c
+    ^-  v-channels:c
+    %-  ~(run by vc)
+    |=  v=v-channel:v7:old:c
+    ^-  v-channel:c
+    =/  [log=log:v-channel:c mod=(map id-post:c time)]
+      (log-7-to-8 log.v)
+    =/  [count=@ud =v-posts:c]
+      (v-posts-7-to-8 posts.v mod)
+    =-  -(|1.- [count |1.-.-])
+    %=  v
+      posts    v-posts
+      log      log
+      future   (future-7-to-8 future.v)
+    ==
+  ++  v-posts-7-to-8
+    |=  [vp=v-posts:v7:old:c mod=(map id-post:c time)]
+    ^-  [@ud v-posts:c]
```

Comment:
I would move the bodies of these arms into /lib/channel-utils, instead of one-to-one duplicating them between channels and channels-server. Just guarantees they'll be consistent/equivalent/the same.